js/controller.js

```diff
@@ -71,8 +71,10 @@
                     console.log("Traffic", durationTraffic);
                     $scope.traffic = {
```

Comment:
If we are going to be adding all this information the promise here should just return the object that's being generated here so you can set it directly onto the scope instead of having to do a transformation here.